File: 프로그래머스/lv2/64065. 튜플/튜플.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public ArrayList solution(String s) {
        ArrayList<Integer> answer=new ArrayList<>();
        s=s.substring(2,s.length());
        s=s.substring(0,s.length()-2);
        s=s.replace("},{"," ");

        String[] str=s.split(" ");
        Arrays.sort(str, new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                return Integer.compare(o1.length(),o2.length());
            }
        });

        for(String K:str){
            String[] tmp=K.split(",");
            for(int i=0;i<tmp.length;i++){
                int n=Integer.parseInt(tmp[i]);
                if(!answer.contains(n)) answer.add(n);
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/64065. 튜플/튜플.py
import re
def solution(s):
    answer = []
    s_arr = s.lstrip('{').rstrip('}').split('},{')
    s_arr.sort(key = len)
    tmp = []
    for s in s_arr:
        tmp.append(s.split(','))
        
    for t in tmp:
        for s in t:
            if int(s) not in answer:
                answer.append(int(s))
            
    return answer